test/tryThis.lean

```diff
@@ -0,0 +1,170 @@
+elab "add_suggestion" s:term : tactic => unsafe do
+  addSuggestion (← getRef) (← evalTerm Suggestion (.const ``Suggestion []) s)
+
+/-- Add a suggestion with a header. -/
+elab "add_suggestion" s:term "with_header" h:term : tactic => unsafe do
+  addSuggestion (← getRef) (← evalTerm Suggestion (.const ``Suggestion []) s)
+    (header := ← evalTerm String (.const ``String []) h)
+
+/-- Add a suggestion. -/
+elab "add_suggestions" s:term : tactic => unsafe do
+  let s ← evalTerm (Array Suggestion) (.app (.const ``Array [.zero]) (.const ``Suggestion [])) s
+  addSuggestions (← getRef) s
+
+/-- Add suggestions with a header. -/
+elab "add_suggestions" s:term "with_header" h:term : tactic => unsafe do
+  let s ← evalTerm (Array Suggestion) (.app (.const ``Array [.zero]) (.const ``Suggestion [])) s
+  addSuggestions (← getRef) s (header := ← evalTerm String (.const ``String []) h)
+
+/-- Demo adding a suggestion. -/
+macro "#demo1" s:term : command => `(example : True := by add_suggestion $s; trivial)
+/--Demo adding a suggestion with a header. -/
+macro "#demo1" s:term "with_header" h:term : command => `(example : True := by
+  add_suggestion $s with_header $h; trivial)
+/--Demo adding suggestions. -/
+macro "#demo" s:term : command => `(example : True := by
+  add_suggestions $s; trivial)
+/--Demo adding suggestions with a header. -/
+macro "#demo" s:term "with_header" h:term : command => `(example : True := by
+  add_suggestions $s with_header $h; trivial)
+
+/-- The syntax you typically get from ``←`(tactic|rfl)`` -/
+private def demoSyntax : TSyntax `tactic := {
+  raw := Lean.Syntax.node
+    (Lean.SourceInfo.none)
+    `Lean.Parser.Tactic.tacticRfl
+    #[Lean.Syntax.atom (Lean.SourceInfo.none) "rfl"] }
+
+/-- A basic suggestion. -/
+private def s : Suggestion := demoSyntax
+
+/-! # Demos -/
+
+-- `Try this: rfl` with `rfl` in text-link color.
+#demo1 s
+
+/-
+```
+Try these:
+• rfl
+• rfl
+• rfl
+• rfl
+```
+with `rfl` in text-link color/ -/
+#demo #[s,s,s,s]
+
+/-
+```
+Try these:
+• rfl -- red
+• rfl -- red-orange
+• rfl -- orange
+• rfl -- yellow
+• rfl -- yellow-green
+• rfl -- light green
+• rfl -- green
+```-/
+#demo #[0.0, 1/6, 2/6, 3/6, 4/6, 5/6, 1.0].map fun t => {s with style? := some <| .value t}
+
+-- `Try this: rfl` -- error color with error squiggle
+#demo1 {s with style? := some .error}
+
+-- `Try this: rfl` -- error color, no squiggle
+#demo1 {s with style? := some <| .error (decorated := false)}
+
+-- `Try this: rfl` -- gold color with warning squiggle
+#demo1 {s with style? := some .warning}
+
+-- `Try this: rfl` -- gold color with no squiggle
+#demo1 {s with style? := some <| .warning (decorated := false)}
+
+-- `Try this: rfl` -- Lean green
+#demo1 {s with style? := some .success}
+
+-- `Try this: rfl` -- styled like a goal hypothesis
+#demo1 {s with style? := some .asHypothesis}
+
+-- `Try this: rfl` -- styled like an inaccessible goal hypothesis
+#demo1 {s with style? := some .asInaccessible}
+
+/--info: Try this: rfl-/
```

Comment:
use the code action to generate the message, it uses better formatting

Reply:
Do you mean by moving the `logInfoAt` to inside the `tryThisProvider`? Or something else?

(There would be a couple issues with moving `logInfoAt`, but mainly, I don't think we have access to the `MessageData` at that point, just the string—and apparently we want to be able to specify and use custom `MessageData`. Not exactly sure what the use case is, though.)

Reply:
I mean delete the docstring on the test and then use `ctrl-.` to make the code action put the docstring back

Reply:
Ok, done in efd52f6.